src/polynomial/poly-grads.jl

```diff
@@ -0,0 +1,386 @@
+import ChainRulesCore: frule, rrule
+using LinearAlgebra
+
+function compute_gn_jac(u_n::AbstractVector{T}) where T
+    n = length(u_n) - 1
+    g_n = zero(u_n)
+    dgdu = fill!(similar(u_n, n + 1, n + 1), zero(T))
```

Comment:
Isn't this something like
```suggestion
    dgdu = zeros(T, n + 1, n + 1)
```
?

Reply:
for 99% of applications its equivalent, but in the case that we need to retain the array type (e.g. `u_n` is a `CuArray`) the `similar` call is necessary.

Reply:
I see.  Maybe drop a comment for future readers? :slightly_smiling_face: